doc/user/flowspec.rst

```diff
@@ -0,0 +1,330 @@
+
+Ensuring that a Flowspec entry has been correctly installed and that incoming
+traffic is policy-routed correctly can be checked like illustrated below. First
+of all, you must check whether the Flowspec entry has been installed or not.
+
+.. code-block:: frr
+
+   CLI# show bgp ipv4 flowspec 5.5.5.2/32
+    BGP flowspec entry: (flags 0x418)
+      Destination Address 5.5.5.2/32
+      IP Protocol = 17
+      Destination Port >= 50 , <= 90
+      FS:redirect VRF RT:255.255.255.255:255
+      received for 18:41:37
+      installed in PBR (match0x271ce00)
+
+This means that the Flowspec entry has been installed in a `Linux Netfilter iptable`
+named `match0x271ce00`. Once you have confirmation it is installed, you can check
+whether you find the associate entry by executing following command. You can also
+check whether incoming traffic has been matched by looking at counter line.
+
+.. code-block:: frr
+
+   CLI# show pbr ipset match0x271ce00
+   IPset match0x271ce00 type net,port
+        to 5.5.5.0/24:proto 6:80-120 (8)
+           pkts 1000, bytes 1000000
+        to 5.5.5.2:proto 17:50-90 (5)
+           pkts 1692918, bytes 157441374
+
+As you can see, the entry is present. note that a `Linux Netfilter iptable can be used
+to host several BGP Flowspec entries. In order to know where the matching traffic is
+redirected to, you have to look at the policy routing rules. The policy-routing is done
+by forwarding traffic to a routing table number. That routing table number is reached by
+using a `Netfilter Linux iptable`. The relationship between the routing table number and
+the incoming traffic is a MARKER that is set by the IPtable referencing the IPSet.
+In Flowspec case, `Linux Netfilter iptable` referencing the `Linux Netfilter ipset` context
+have the same name. So it is easy to know which routing table is used by issuing following
+command:
+
+.. code-block:: frr
+
+   CLI# show pbr iptable
+      IPtable match0x271ce00 action redirect (5)
+        pkts 1700000, bytes 158000000
+        table 257, fwmark 257
+   ...
+
+As you can see, by using following Linux commands, the MARKER `0x101` is present in
+both IPtables and IPRule contexts.
+
+.. code-block:: shell
+
+   # iptables -t mangle --list match0x271ce00 -v
+   Chain match0x271ce00 (1 references)
+   pkts bytes target     prot opt in     out     source               destination
+   1700K  158M MARK       all  --  any    any     anywhere             anywhere
+        MARK set 0x101
+   1700K  158M ACCEPT     all  --  any    any     anywhere             anywhere
+
+   # ip rule list
+   0:from all lookup local
+   0:from all fwmark 0x101 lookup 257
+   32766:from all lookup main
+   32767:from all lookup default
+
+This allows us to see where the traffic is forwarded to.
+
+Limitations / Known issues
+--------------------------
+
+As you can see, Flowspec is rich and can be very complex.
+As of today, not all flowspec rules will be able to be converted into Policy
```

Comment:
Capitalize `Flowspec`, and either lowercase `Policy` or or make it `Policy Based Routing`.

Reply:
Policy Based Routing.

Reply:
Capitalize `Flowspec` please.